src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -196,7 +177,7 @@ internal sealed class DirectoryLoadContext : AssemblyLoadContext
             private readonly AnalyzerAssemblyLoader _loader;
 
             public DirectoryLoadContext(string directory, AnalyzerAssemblyLoader loader)
-                : base(isCollectible: true)
+                : base(isCollectible: false)
```

Comment:
@CyrusNajmabadi I saw that Extensions use the assembly loader. Will this break those scenarios?

Reply:
Opened https://github.com/dotnet/roslyn/pull/79997